customviews/ListItem.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import {StyleSheet, Text, TouchableHighlight, View} from "react-native";
+
+export class ListItem extends React.Component {
+
+    constructor(props){
+        super(props);
+    }
+
+    render(){
+        return (
+            <TouchableHighlight onPress={()=>this.props.context.navigation.navigate('Details', {item: this.props.item})}>
```

Comment:
Try not not define function inline. It provides to redeclaring function (in js is not very cheap) on every rerender or refresh on view. 

But it's not very important:

```
doSth = () => { /* */ }
// or
doSthEls() { /* */ }
// ...
render() {
	return(
		<TouchableOpacity onPres={this.doSth} />
	)
} 


```